code/game/objects/items/devices/portable_recharger.dm

```diff
@@ -0,0 +1,330 @@
+		. += span_notice("The status display reads:")
+
+	var/obj/item/stock_parts/cell/charging_cell = charging.get_cell()
+	if(charging_cell)
+		. += span_notice("- \The [charging]'s cell is at <b>[charging_cell.percent()]%</b>.")
+		return
+	if(istype(charging, /obj/item/ammo_box/magazine/recharge))
+		var/obj/item/ammo_box/magazine/recharge/power_pack = charging
+		. += span_notice("- \The [charging]'s cell is at <b>[PERCENT(power_pack.stored_ammo.len/power_pack.max_ammo)]%</b>.")
+		return
+	. += span_notice("- \The [charging] is not reporting a power level.")
+
+/obj/item/portable_recharger/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	if(is_type_in_typecache(arrived, allowed_devices))
+		charging = arrived
+		START_PROCESSING(SSmachines, src)
+		finished_recharging = FALSE
+		using_power = TRUE
+		soundloop.start()
+		update_appearance()
+	return ..()
+
+/obj/item/portable_recharger/Exited(atom/movable/gone, direction)
+	if(gone == charging)
+		if(!QDELING(charging))
+			charging.update_appearance()
+		charging = null
+		using_power = FALSE
+		soundloop.stop()
+		update_appearance()
+	return ..()
+
+/obj/item/portable_recharger/attackby(obj/item/attacking_item, mob/user, params)
+	if(hit)
+		return
+
+	if(istype(attacking_item, /obj/item/stock_parts/cell))
+		if(cell)
+			to_chat(user, span_warning("[src] already has a cell!"))
+		else
+			if(!user.transferItemToLoc(attacking_item, src))
+				return TRUE
+			cell = attacking_item
+			to_chat(user, span_notice("You install a cell in [src]."))
+
+	if(!cell)
+		balloon_alert(user, "no cell!")
+		return ..()
+
+	if(!is_type_in_typecache(attacking_item, allowed_devices))
+		return ..()
+
+	if(charging)
+		return TRUE
+
+	if(!cell.charge)
+		to_chat(user, span_notice("[src] blinks red as you try to insert [attacking_item]."))
+		return TRUE
+
+	if(istype(attacking_item, /obj/item/gun/energy))
+		var/obj/item/gun/energy/energy_gun = attacking_item
+		if(!energy_gun.can_charge)
+			to_chat(user, span_notice("Your gun has no external power connector."))
+			return TRUE
+	user.transferItemToLoc(attacking_item, src)
+	return TRUE
+
+/obj/item/portable_recharger/screwdriver_act(mob/living/user, obj/item/tool)
+	if(!cell || charging || hit)
+		return FALSE
+
+	if(do_after(user, 3 SECONDS, target = src))
+		cell.update_appearance()
+		cell.forceMove(get_turf(src))
+		balloon_alert(user, "removed [cell]")
+		cell = null
+		tool.play_tool_sound(src, 50)
+		return TRUE
+	return FALSE
+
+/obj/item/portable_recharger/attack_hand(mob/user, list/modifiers)
+	if(loc == user)
+		if(slot_flags & ITEM_SLOT_BACK)
+			if(user.get_item_by_slot(ITEM_SLOT_BACK) == src)
+				take_charging_out(user)
+			else
+				to_chat(user, span_warning("Put the [src] on your back first!"))
+
+		else if(slot_flags & ITEM_SLOT_BELT)
+			if(user.get_item_by_slot(ITEM_SLOT_BELT) == src)
+				take_charging_out(user)
+			else
+				to_chat(user, span_warning("Strap the [src] on first!"))
+		return
```

Comment:
I recognize this code - and if there's anything you shouldn't copy+paste, it's defib code
```suggestion
		if(user.get_slot_by_item(src) & slot_flags)
			take_charging_out(user)
		else
			balloon_alert(user, "equip it first!")
		return TRUE
```